lib/Compilation.js

```diff
@@ -362,13 +362,13 @@ class Compilation extends Tapable {
 	_addModuleChain(context, dependency, onModule, callback) {
 		const start = this.profile && Date.now();
 
-		const errorAndCallback = this.bail ? function errorAndCallback(err) {
+		const errorAndCallback = this.bail ? (err) => {
 			callback(err);
-		} : function errorAndCallback(err) {
```

Comment:
Are you referring so it stays in the stack trace?  Did a little digging and it looks like it shows up when assigned to a variable? 

http://lukecod.es/2015/12/01/es6-arrow-function-stack-traces/

https://stackoverflow.com/questions/27977525/how-do-i-write-a-named-arrow-function-in-es2015#answers-header